### DISCUSSION TITLE
issue: Validation fails on non-required fields

### DISCUSSION THREAD

**Discussion by 5c077m4n** [react-hook-form/react-hook-form#7304] [Q&A]
### Version Number

7.22.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-non-required-validation-bug-pjys9?file=/src/App.js

### Steps to reproduce

1. Go to my [demo](https://codesandbox.io/s/react-hook-form-non-required-validation-bug-pjys9?file=/src/App.js)
2. Once pressing the `submit` button you can see that the not required file field fails the validation - even though it's not required

### Expected behaviour

The form should submit.

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
check your validate function, if it returns false it's considered invalid.

**Reply by 5c077m4n** [reply]
Thank you for the super quick answer!
But if the required field is false and the value undefined I would expect the validation test to pass (so it wouldn't run at all - just automatically return true)

**Reply by 5c077m4n** [reply]
I forgot to mention that if I do select a file then the form is valid
